rskj-core/src/test/java/co/rsk/peg/pegininstructions/PeginInstructionsVersion1Test.java

```diff
@@ -0,0 +1,109 @@
+        peginInstructionsVersion1.validateDataLength(Hex.decode("00010e537aad84447a2c2a7590d5f2665ef5cf9b667a98"));
+    }
+
+    @Test
+    public void peginInstructionsV1_getProtocolVersion() {
+        PeginInstructionsVersion1 peginInstructionsVersion1 = new PeginInstructionsVersion1(params);
+        Assert.assertEquals(1, peginInstructionsVersion1.getProtocolVersion());
+    }
+
+    @Test
+    public void peginInstructionsV1_getRskDestinationAddress_no_data() {
+        PeginInstructionsVersion1 peginInstructionsVersion1 = new PeginInstructionsVersion1(params);
+        Assert.assertNull(peginInstructionsVersion1.getRskDestinationAddress());
+    }
+
+    @Test(expected = PeginInstructionsParseException.class)
+    public void peginInstructionsV1_parse_data_invalid_length() throws PeginInstructionsParseException {
+        PeginInstructionsVersion1 peginInstructionsVersion1 = new PeginInstructionsVersion1(params);
+        peginInstructionsVersion1.parse(Hex.decode("00010e537aad84447a2c2a7590d5f2665ef5cf9b"));
+    }
+
+    @Test
+    public void peginInstructionsV1_getBtcRefundAddressFromData_no_refund_address() throws
+            PeginInstructionsParseException {
+        PeginInstructionsVersion1 peginInstructionsVersion1 = new PeginInstructionsVersion1(params);
+
+        Assert.assertEquals(Optional.empty(),
+                peginInstructionsVersion1.getBtcRefundAddressFromData(
+                        Hex.decode("00010e537aad84447a2c2a7590d5f2665ef5cf9b667a")));
+    }
+
+    @Test
+    public void peginInstructionsV1_getBtcRefundAddressFromData_p2pkh_address() throws
+            PeginInstructionsParseException {
+        PeginInstructionsVersion1 peginInstructionsVersion1 = new PeginInstructionsVersion1(params);
+
+        Address btcAddress = new Address(params, Hex.decode("4f4c767a2d308eebb3f0f1247f9163c896e0b7d2"));
+
+        String rawData = "00010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2";
+        Optional<Address> obtainedBtcAddress = peginInstructionsVersion1.getBtcRefundAddressFromData(Hex.decode(rawData));
+
+        Assert.assertEquals(Optional.of(btcAddress), obtainedBtcAddress);
+    }
+
+    @Test
+    public void peginInstructionsV1_getBtcRefundAddressFromData_p2sh_p2wpkh_address() throws
+            PeginInstructionsParseException {
+        PeginInstructionsVersion1 peginInstructionsVersion1 = new PeginInstructionsVersion1(params);
+
+        Address btcAddress = new Address(params,
+                params.getP2SHHeader(),
+                Hex.decode("19d7e0ee9bf6bd70d1d046b066d1c2726e1accc1"));
+
+        String rawData = "00010e537aad84447a2c2a7590d5f2665ef5cf9b667a0219d7e0ee9bf6bd70d1d046b066d1c2726e1accc1";
+        Optional<Address> obtainedBtcAddress = peginInstructionsVersion1.getBtcRefundAddressFromData(Hex.decode(rawData));
+
+        Assert.assertEquals(Optional.of(btcAddress), obtainedBtcAddress);
```

Comment:
Assert.assertTrue(obtainedBtcAddress.isP2SHAddress());

Reply:
Check that the obtainedBtcAddress is P2SH not the one you created

Reply:
That's true!